pkg/datapath/linux/node.go

```diff
@@ -112,6 +114,19 @@ func NewNodeHandler(
 	}
 
 	handler := newNodeHandler(log, datapathConfig, nodeMap, nodeManager)
+
+	lifecycle.Append(cell.Hook{
+		OnStart: func(_ cell.HookContext) error {
+			handler.RestoreNodeIDs()
+			nodeManager.Subscribe(handler)
```

Comment:
I would consider doing the `Subscribe` already inside `NewNodeHandler`. This way the subscription is set up before anything has started.

Reply:
Thanks, good point -> refactored.

(At first i thought that this might results in errors during `RestoreNodeIDs` checking for already existing node ids. (i faced this one first) But this shouldn't occur in this case as the subscription shouldn't add any nodes before startup :crossed_fingers: ))